### PR TITLE
frontend: [shared] Transfer diverged changes from gitlab to github

### PR DESCRIPTION
**Pull request by gorankokin**
* Why:
   * It's needed to sync both repositories and start backporting open sourced libraries to other internal projects
* What:
   * Transfering changes
* Testing:
   * Linting, unit tests, build lib/ui locally and e2e tests in gitlab
* Change:
   * Major, there are some breaking changes in shared repository

Formatting changes are included because without them I had issue to effectively transfer my changes from gitlab to github due to my custom intellij setup that auto-apply copyright against pre-defined rules.
formatting changes are quite small to include them in another MR.
see: https://sass-lang.com/documentation/syntax/comments

### CODE REVIEW

**Comment by ivakoleva** [comment]
Please rename to frontend prefix? It is dedicated to frontend/ components here https://github.com/vmware/versatile-data-kit/blob/main/.gitlint#L9 

The commit message format to use is https://github.com/vmware/versatile-data-kit/blob/main/support/git-commit-template.txt

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -104,13 +104,21 @@ repos:
           - --comment-style
           - <!--|   ~|  -->
       - id: insert-license
```
Comment (ivakoleva) [inline .pre-commit-config.yaml:106]: Please split formatting changes in separate PR, unless there is an explicit reason not to (in case there is, please add to PR description)

File: projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss [hunk 2]
```diff
@@ -1,3 +1,8 @@
+/*!
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
 /*
  * Copyright 2021-2023 VMware, Inc.
  * SPDX-License-Identifier: Apache-2.0
```
Comment (ivakoleva) [inline projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss:8]: Please cleanup from duplicates?

**Comment by ivakoleva** [comment]
To Testing Done, may you add the local tests performed if any? For example, listed, built, run unit tests, started the gui

File: projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss (hunk 2 above)
Comment (ivakoleva) [reply projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss:8]: Seems to be due the license pre-commit hook changes made in the scope of this PR to me

Comment (gorankokin) [reply projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss:3]: Yes exactly, it's because pre-commit config changes from my side and they pop-up from the other library data-pipelines, but local pre-commit didn't catch them because I didn't have changes there

Comment (gorankokin) [reply projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss:3]: I fix them and there is amend to existing commit and squash everything into 1 commit

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -104,13 +104,21 @@ repos:
...
           - --comment-style
           - /*| *| */
+      - id: insert-license
+        files: \.scss
+        args:
+          - --use-current-year
+          - --license-filepath
+          - NOTICE.txt
+          - --comment-style
+          - /*!| *| */
```
Comment (ivakoleva) [inline .pre-commit-config.yaml:121]: I don't understand the need for changing the hook config. What is the purpose for changing the comment style for scss files? What problem does it solve?

File: projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/*
+/*!
```
Comment (DeltaMichael) [inline projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/base-grid/_data-jobs-base-grid.component.scss:1]: How did that exclamation point get there? :D

File: projects/frontend/shared-components/gui/.editorconfig [hunk 5]
```diff
@@ -3,11 +3,14 @@ root = true
 
 [*]
 charset = utf-8
-indent_style = space
+indent_style = tab
```
Comment (DeltaMichael) [inline projects/frontend/shared-components/gui/.editorconfig:6]: Why change this? I think spaces is better, due to some operating systems/editors handling tabs differently.

File: projects/frontend/shared-components/gui/projects/shared/src/test.ts [hunk 6]
```diff
@@ -25,13 +22,9 @@ declare const require: {
...
 
 // First, initialize the Angular testing environment.
-getTestBed().initTestEnvironment(
-	BrowserDynamicTestingModule,
-	platformBrowserDynamicTesting(),
-	{
-		teardown: { destroyAfterEach: false }
-	}
-);
+getTestBed().initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting(), {
```
Comment (DeltaMichael) [inline projects/frontend/shared-components/gui/projects/shared/src/test.ts:25]: A lot of the changes seem to be reformatting of existing files. This is probably due to the linter changes. Not sure this is even a problem, just pointing it out. 🙂

File: projects/frontend/shared-components/gui/.editorconfig (hunk 5 above)
Comment (gorankokin) [reply projects/frontend/shared-components/gui/.editorconfig:6]: It was with space in .editorconfig, but prettier that is used as plugin in pre-commit, was using tabs in its own config, and pre-hook auto-format was applying tabs instead of spaces.
I prefer spaces too, but that will have implications on all files, because plugin was not configured properly before.
see file: `projects/frontend/shared-components/gui/.prettierrc.json`

File: projects/frontend/shared-components/gui/projects/shared/src/test.ts (hunk 6 above)
Comment (gorankokin) [reply projects/frontend/shared-components/gui/projects/shared/src/test.ts:25]: I was trying to align code formatting with existing formatting in taurus and supercollider.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (gorankokin) [reply .pre-commit-config.yaml:121]: I changed that because when I use copyright auto update from Intellij it append copyright with format that I specify in the .pre-commit-config.yaml which is described in sass documentation that is comment that exists in the file after transpiled to css.

```
// This comment won't be included in the CSS.

/* But this comment will, except in compressed mode. */

/* It can also contain interpolation:
 * 1 + 1 = #{1 + 1} */

/*! This comment will be included even in compressed mode. */
```

File: projects/frontend/shared-components/gui/projects/shared/src/test.ts (hunk 6 above)
Comment (gorankokin) [reply projects/frontend/shared-components/gui/projects/shared/src/test.ts:25]: @DeltaMichael If you agree I could change to spaces and commit whole project in this commit, or we could do it in separate commit and Pull request.

Taurus and Supercollider are working with indent with spaces, 4 for *.ts/*.js and 2 for *.json and also we removed prettier from the projects, because auto formatting is not very good in some cases, and manual formatting is much better and readable. 
To avoid big code style format differences we extracted one configuration file for Intellij that is somewhere in confluence and is a start point that is used by everyone that contribute in UI projects, so we use Intellij auto-format plus manual formatting in some places and that's how we were working in the last years.

File: projects/frontend/shared-components/gui/.editorconfig (hunk 5 above)
Comment (gorankokin) [reply projects/frontend/shared-components/gui/.editorconfig]: I just revert it back with space and I committed all shared-components directory files with the new formatting with spaces.

File: projects/frontend/shared-components/gui/projects/shared/src/test.ts (hunk 6 above)
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/test.ts:25]: I'm fine with leaving it as is if that's easier. @ivakoleva Are `*.ts` files included in the GitHub format action?

Comment (gorankokin) [reply projects/frontend/shared-components/gui/projects/shared/src/test.ts:25]: I formatted the whole project `shared-components` with indent space, where 4 spaces for .ts/.js and 2 spaces for *.json, and force prettier to leverage .editorconfig as a source of truth to make sync between our local IDEs and github

**Review by DeltaMichael — APPROVED** [review]

**Comment by ivakoleva** [comment]
This PR is approved, will be merged in the scope of https://github.com/vmware/versatile-data-kit/pull/1746 (since releases are co-dependent)
